js.go

```diff
@@ -826,6 +857,67 @@ type jsSub struct {
 	cmeta atomic.Value
 }
 
+// newFetchReply generates a unique inbox used for a fetch request.
+func (jsi *jsSub) newFetchReply() string {
+	jsi.mu.Lock()
+	rpre := jsi.rpre
+	rn := jsi.rr.Int63()
+	jsi.mu.Unlock()
+	var sb strings.Builder
+	sb.WriteString(rpre)
+	var b [aReplyTokensize]byte
+	for i, l := 0, rn; i < len(b); i++ {
+		b[i] = rdigits[l%base]
+		l /= base
+	}
+	sb.Write(b[:])
+	return sb.String()
+}
+
+// handleFetch is delivered a message requested by pull subscribers
+// when calling Fetch.
+func (jsi *jsSub) handleFetch(m *Msg) {
+	jsi.mu.Lock()
+	if len(jsi.freqs) == 0 {
+		jsi.mu.Unlock()
+		return
+	}
+	mch := jsi.freqs[0]
+	jsi.freqs = append(jsi.freqs[:0], jsi.freqs[1:]...)
```

Comment:
Check if we have more than 1, if not optimize that case with a reslice to [:0]

Reply:
Updated this one too